api/src/test/java/com/epam/pipeline/manager/HierarchicalEntityManagerTest.java

```diff
@@ -16,348 +16,312 @@
+
+        assertThat(available.size()).isEqualTo(1);
+        assertThat(available.get(AclClass.CONFIGURATION).get(0).getMask()).isEqualTo(READ_PERMISSION);
     }
 
     @Test
-    @WithMockUser(username = USER)
-    public void testUseItOwnPermissionsIfParentRestrictedByInheritance() {
-        Folder folder = createFolder(TEST_NAME, null);
-        Folder subFolder = createFolder(TEST_NAME, folder.getId());
-        RunConfiguration runConfiguration = createRunConfiguration(TEST_NAME, subFolder.getId());
-
-        grantPermission(folder.getId(), AclClass.FOLDER, USER2, true, AclPermission.NO_READ.getMask());
-        grantPermission(runConfiguration.getId(), AclClass.CONFIGURATION, USER2, true,
-                ALL_PERMISSIONS);
-
-        Map<AclClass, List<AbstractSecuredEntity>> available = hierarchicalEntityManager
-                .loadAvailable(new AclSid(USER2, true), null);
-        //folder should be filtered because it forbidden
-        Assert.assertEquals(1, available.size());
-        Assert.assertEquals((int) available.get(AclClass.CONFIGURATION).get(0).getMask(), ALL_PERMISSIONS_SIMPLE);
-    }
-
-    @Test
-    @WithMockUser(username = USER)
+    @WithMockUser
     public void shouldReturnOneAclClassOnly() {
-        Folder folder = createFolder(TEST_NAME, null);
-        Folder subFolder = createFolder(TEST_NAME, folder.getId());
-        RunConfiguration runConfiguration = createRunConfiguration(TEST_NAME, subFolder.getId());
-
-        grantPermission(folder.getId(), AclClass.FOLDER, USER2, true, ALL_PERMISSIONS);
-        grantPermission(runConfiguration.getId(), AclClass.CONFIGURATION, USER2, true,
-                ALL_PERMISSIONS);
-
-        Map<AclClass, List<AbstractSecuredEntity>> available = hierarchicalEntityManager
-                .loadAvailable(new AclSid(USER2, true), AclClass.CONFIGURATION);
-        Assert.assertEquals(1, available.size());
-        Assert.assertEquals((int) available.get(AclClass.CONFIGURATION).get(0).getMask(), ALL_PERMISSIONS_SIMPLE);
+        final Folder root = new Folder();
+        final Folder folder = getFolder(ANOTHER_SIMPLE_USER);
+        final Folder childFolder = getFolder(ID_2, folder.getId(), ANOTHER_SIMPLE_USER);
+        final RunConfiguration runConfiguration = getRunConfiguration(ID_2, ANOTHER_SIMPLE_USER);
+        childFolder.setConfigurations(Collections.singletonList(runConfiguration));
+        root.setChildFolders(Collections.singletonList(folder));
+        folder.setChildFolders(Collections.singletonList(childFolder));
+        doReturn(root).when(mockFolderManager).loadTree();
+        mockDockerRegistryList();
+        mockUserContext();
+        initAclEntity(folder, AclPermission.READ);
+        initAclEntity(runConfiguration, AclPermission.READ);
+
+        final Map<AclClass, List<AbstractSecuredEntity>> available = hierarchicalEntityManager
+                .loadAvailable(new AclSid(SIMPLE_USER, true), AclClass.CONFIGURATION);
```

Comment:
Maybe it would be cleaner to move `new AclSid(SIMPLE_USER, true)` to test class constants:
```
private static final AclSid USER_SID = new AclSid(SIMPLE_USER, true);
```
?